diff --git a/board/phytec/phycore_imx8mp/phycore_imx8mp.env b/board/phytec/phycore_imx8mp/phycore_imx8mp.env
index fd1bd95a60e..e1c02f32ea6 100644
--- a/board/phytec/phycore_imx8mp/phycore_imx8mp.env
+++ b/board/phytec/phycore_imx8mp/phycore_imx8mp.env
@@ -2,6 +2,7 @@
 #include <env/phytec/overlays.env>
 #include <env/phytec/fitboot.env>

+optargs=exampleVar=1
 bootcmd=
	if test ${dofastboot} = 1; then
		fastboot 0;
